extern/warprnnt/__init__.py

```diff
@@ -0,0 +1,103 @@
+"""
+Provides a RETURNN wrapper around `warp-transducer`:
+  https://github.com/HawkAaron/warp-transducer.git
+
+Importing this module immediately compiles the library and TF module.
+"""
+
+import tensorflow as tf
+from tensorflow.python.framework import ops
+import os
+from TFUtil import OpCodeCompiler
+
+
+warprnnt_dir = os.path.dirname(os.path.abspath(__file__))
+submodule_dir = os.path.join(warprnnt_dir, "warp-transducer")
+_tf_mod = None
+
+
+def is_checked_out():
+  """Checks if the git submodule is checkout out."""
+  return os.path.isfile("%s/src/rnnt_entrypoint.cpp" % submodule_dir)
+
+
+def init_warprnnt(verbose=False):
+  """
+  Initialiazes and compiles the library. Caches the TF module.
+  :param bool verbose:
+  """
+  global _tf_mod
+  assert is_checked_out(), "submodule not checked out? Run `git submodule update --init --recursive`"
+
+  # References:
+  # https://github.com/HawkAaron/warp-transducer/blob/master/tensorflow_binding/setup.py
+
+  src_files = ['%s/src/rnnt_entrypoint.cpp' % submodule_dir,
+               '%s/tensorflow_binding/src/warprnnt_op.cc' % submodule_dir]
+  assert all([os.path.isfile(f) for f in src_files]), "submodule in %r not checked out?" % warprnnt_dir
+  src_code = ""
+  for fn in src_files:
+    f_code = open(fn).read()
+    src_code += "\n// ------------ %s : BEGIN { ------------\n" % os.path.basename(fn)
+    # https://gcc.gnu.org/onlinedocs/cpp/Line-Control.html#Line-Control
+    src_code += "#line 1 \"%s\"\n" % os.path.basename(fn)
+    src_code += f_code
+    src_code += "\n// ------------ %s : END } --------------\n\n" % os.path.basename(fn)
+
+  compiler = OpCodeCompiler(
+    base_name="warprnnt_kernels", code_version=1, code=src_code,
+    include_paths=(submodule_dir + "/include",),
+    c_macro_defines={"WITH_OMP": 1},
+    ld_flags=["-Xcompiler", "-fopenmp"],
```

Comment:
It uses OMP? But it is only a GPU implementation?

Reply:
It should work on both CPU and GPU, but I only tested the GPU version.
Their cmake script used omp, so I just added it aswell.

Reply:
Oh interesting. Can you just try out? It should already work, or not?